primitives/runtime/src/lib.rs

```diff
@@ -944,6 +944,38 @@ impl<R> TransactionOutcome<R> {
 	}
 }
 
+/// Confines the logic that can be executed in a block.
+///
+/// The runtime may refuse to import blocks that violate these requirements.
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Encode, Decode, TypeInfo)]
+pub enum RuntimeExecutiveMode {
+	/// All logic is allowed to run.
+	///
+	/// For example:
+	/// - Extrinsics with dispatch class `Normal`.
+	/// - `on_idle` hook.
+	Normal,
+	/// Only _necessary_ logic is allowed to run.
+	///
+	/// Explicitly forbidden are:
+	/// - Extrinsics with dispatch classes `Normal` and `Operational`.
+	/// - `on_idle` and `poll` hooks.
+	///
+	/// Explicitly allowed are:
+	/// - Mandatory extrinsics (i.e. via OCW).
+	/// - `on_initialize` and `on_finalize` hooks.
+	/// - Storage migrations.
+	///
+	/// Everything in between is to be judged by the runtime.
+	Minimal,
```

Comment:
I would probably call it `OnlyInherents` and `AllExtrinsics` which is more correct?

Also `RuntimExecutionMode` is a weird name.

Reply:
`ExtrinsicInclusionMode`?

Reply:
I do like `Normal` and `Minimal` (or could be `System`). it's not only about non-inherents inclusion, but also about hooks and could be more in the future. More generic names with documentation looks like more flexible approach here. My five cents.

Reply:
But this is here about communicating to the node. The node doesn't know anything about a poll mode nor does it cares. You are working here on the boundary between runtime and node. The runtime is opaque to the node and the node only has certain runtime apis to talk to the runtime.

Reply:
Oh i mixed that up. Will change it in the RFC.